futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/GroupDataProvider.cs

```diff
@@ -461,5 +461,41 @@ public async Task<GroupSite> GetGroupSiteDataAsync(string groupSlug, Cancellatio
 
             return groupSiteData;
         }
+
+        public async Task<bool> GetGroupPrivacyStatusAsync(string groupSlug, CancellationToken cancellationToken = default)
+        {
+            const string query =
+                             @$"SELECT
+                                    PublicGroup
+                               FROM[Group]
+                               WHERE Slug = @Slug";
+
+            using var dbConnection = await _connectionFactory.GetReadOnlyConnectionAsync(cancellationToken);
+
+            var isPublicGroup = await dbConnection.QueryFirstOrDefaultAsync<bool>(query, new
```

Comment:
Should be `QuerySingleOrDefaultAsync` as we're only expecting to have one group per slug, same for groupId below